File: design-pattern-go/src/patterns/behavioral/chain_responsibility/Container.go
package chainresponsibility

type Container struct {
	ComponentWithHandler
	Leaf
}

func NewContainer(handler func()) Container {
	return Container{
		Leaf: NewLeaf(handler),
	}
}

func (c *Container) Add(leaf *Leaf) *Container {
	leaf.SetContainer(c)
	return c
}
func (c *Container) Handle() {
	c.Leaf.Handle()
}

File: design-pattern-jvm/app/src/main/java/design/pattern/java/patterns/behavioral/chain_responsibility/Container.java
package design.pattern.java.patterns.behavioral.chain_responsibility;

import java.util.function.Supplier;

public class Container extends Leaf {
  public Container(Supplier<Boolean> handler) {
    super(handler);
  }

  public Container add(Leaf leaf) {
    leaf.setContainer(this);
    return this;
  }

}
